src/fs.register.ts

```diff
@@ -31,8 +32,40 @@ export const fqdn: FinalizeRequestMiddleware<object, MetadataBearer> = (next) =>
   };
 };
 
+/**
+ * AWS SDK middleware logic to try 3 times if receiving an EAI_AGAIN error
+ */
+export function eaiAgainBuilder(timeout: (attempt: number) => number): BuildMiddleware<object, MetadataBearer> {
+  const eaiAgain: BuildMiddleware<object, MetadataBearer> = (next) => {
+    const maxTries = 3;
+    return async (args) => {
+      for (let attempt = 1; attempt <= maxTries; attempt++) {
+        try {
+          return await next(args);
+        } catch (error) {
+          if (error && typeof error === 'object' && 'code' in error && 'hostname' in error) {
+            if (error.code !== 'EAI_AGAIN') {
+              throw error;
+            }
+            logger.warn({ host: error.hostname }, `eai_again retry #${attempt}`);
+            await setTimeout(timeout(attempt));
+          } else {
+            throw error;
+          }
+        }
+      }
+      throw new Error(`EAI_AGAIN maximum tries (${maxTries}) exceeded`);
+    };
+  };
+  return eaiAgain;
+}
+
 const client = new S3Client();
 export const s3Fs = new FsAwsS3V3(client);
+client.middlewareStack.add(
+  eaiAgainBuilder((attempt: number) => attempt * 1000),
```

Comment:
`0 * 1000 = 0` which is a instant retry and possibly a instant failure, we should set a minimum delay

```suggestion
  eaiAgainBuilder((attempt: number) => 100 + attempt * 1000),
```